### DISCUSSION TITLE
Any diagrams to better understand your system?

### DISCUSSION THREAD

**Discussion by abubake** [Robotic-Decision-Making-Lab/blue#132] [General]
Hi Evan, I was wondering if you have created any diagrams or ros graphs that give a visual understanding of the structure of your code. If not, could be good for your documentation :)

**Comment by evan-palmer** [comment]
Hey @abubake! Thanks for checking out the project! I do have some diagrams that I can upload. I'll need to clean them up a bit first, but I will share those here as soon as I finish that. The diagrams that I do have include a deployment diagram (to show how to deploy this on a robot) and a class diagram. Are there any other diagrams that would be helpful for you?

**Comment by abubake** [comment]
Hi Evan,

That would be great! What about a ros graph to show the connection between
the nodes/packages? Or something similar.

Best,
Baker Herrin

**Comment by timmarkhuff** [comment]
I second this request. My team has been struggling to get things running. We are making a few assumptions about how it is supposed to work, and are probably just doing something wrong. Any documentation would be appreciated.

**Comment by evan-palmer** [comment]
I've started to put together some documentation. I'll be updating the [gh-pages](https://github.com/Robotic-Decision-Making-Lab/blue/tree/gh-pages) branch as I go. The site will be deployed to https://robotic-decision-making-lab.github.io/blue/ (it's technically deployed there already, but you won't see anything useful yet as I haven't fixed the build pipelines for it). If you would like to build the documentation from source, you just need Node.js >= 16.14, and you can run
```
npm start
```
from the root directory

**Comment by evan-palmer** [comment]
After much procrastination, I've finished up some documentation that I plan to merge as part of [33-feature-ros2-control](https://github.com/Robotic-Decision-Making-Lab/blue/tree/33-feature-ros2-control). You can find the documentation [here](https://robotic-decision-making-lab.github.io/blue/)
